src/solver/pooling/backwardNd.cpp

```diff
@@ -54,165 +69,180 @@ PoolingBackwardNd::GetSolution(const ExecutionContext&,
 {
     auto result = ConvSolution{miopenStatusSuccess};
 
+    auto kernel        = KernelInfo{};
+    kernel.kernel_file = "MIOpenPoolingBwdND.cl";
+    kernel.kernel_name = "mloPoolingND";
+
+    if(problem.GetPooling().GetMode() == miopenPoolingMax)
+    {
+        kernel.kernel_name += "MaxBwd";
+    }
+    else if(problem.GetPooling().GetMode() == miopenPoolingAverage ||
+            problem.GetPooling().GetMode() == miopenPoolingAverageInclusive)
     {
-        auto kernel = KernelInfo{};
+        kernel.kernel_name += "AveBwd";
+    }
 
-        kernel.kernel_file = "MIOpenPoolingBwdND.cl";
-        kernel.kernel_name = "mloPoolingND";
+    const auto bot = problem.GetXDesc();
+    const auto top = problem.GetYDesc();
```

Comment:
```suggestion
    const auto& bot = problem.GetXDesc();
    const auto& top = problem.GetYDesc();
```
Shall it be a reference to avoid extra copy?

Reply:
(Let me check tomorrow)

Reply:
[resolved]